File: HPWorkpath_v1.6.3/Samples/ExampleExtensions/source/GoogleSigninSample/src/com/hp/workpath/sample/googlesigninsample/FileListActivity.java
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.googlesigninsample;

import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;
import android.view.View;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.google.api.services.drive.model.File;
import com.hp.workpath.sample.googlesigninsample.adapter.FileListViewAdapter;
import com.hp.workpath.sample.googlesigninsample.task.DriveFileListTask;

import java.util.List;

public class FileListActivity extends AppCompatActivity {

    private static final String TAG = MainActivity.TAG;

    public static final String INTENT_CREDENTIAL = "intent_credential";

    private ListView mFileLisView;
    private TextView mNoFileTextView;
    private FileListViewAdapter mAdapter;
    private String mCredential;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_file_list);
        initView();
        mCredential = getIntent().getStringExtra(INTENT_CREDENTIAL);
    }

    private void initView() {
        mFileLisView = findViewById(R.id.file_list_view);
        mAdapter = new FileListViewAdapter(this);
        mFileLisView.setAdapter(mAdapter);
        mNoFileTextView = findViewById(R.id.no_file_text_view);
    }

    @Override
    protected void onPause() {
        super.onPause();
    }

    @Override
    protected void onResume() {
        super.onResume();
        if (!TextUtils.isEmpty(mCredential)) {
            new DriveFileListTask(this, mCredential, taskInterface).taskExecute();
        }
    }

    DriveFileListTask.DriveTaskInterface taskInterface = new DriveFileListTask.DriveTaskInterface() {
        @Override
        public void onFailure(Throwable t) {
            if (t != null && TextUtils.isEmpty(t.getMessage())) {
                Log.e(TAG, "Load failure: " + t.getMessage());
                Toast.makeText(FileListActivity.this, "Load failure: " + t.getMessage(), Toast.LENGTH_SHORT).show();
            }
        }

        @Override
        public void onResponse(List<File> files) {
            if (files == null || files.size() == 0) {
                mNoFileTextView.setVisibility(View.VISIBLE);
            } else {
                mAdapter.setItems(files);
                mAdapter.notifyDataSetChanged();
            }
        }
    };
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleExtensions/source/GoogleSigninSample/src/com/hp/workpath/sample/googlesigninsample/FileListActivity.kt
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.googlesigninsample

import android.os.Bundle
import android.text.TextUtils
import android.util.Log
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import com.google.api.services.drive.model.File
import com.hp.workpath.sample.googlesigninsample.adapter.FileListViewAdapter
import com.hp.workpath.sample.googlesigninsample.databinding.ActivityFileListBinding
import com.hp.workpath.sample.googlesigninsample.task.DriveFileListTask
import com.hp.workpath.sample.googlesigninsample.task.DriveFileListTask.DriveTaskInterface
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class FileListActivity : AppCompatActivity() {
    private var mCredential: String? = null
    private lateinit var mBindingFileListActivity: ActivityFileListBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBindingFileListActivity = ActivityFileListBinding.inflate(layoutInflater)
        setContentView(mBindingFileListActivity.root)
        mCredential = intent.getStringExtra(INTENT_CREDENTIAL)
    }

    override fun onResume() {
        super.onResume()
        mCredential?.let {
            lifecycleScope.launch(Dispatchers.IO) {
                DriveFileListTask(this@FileListActivity, it, taskInterface).execute()
            }
        }
    }

    private var taskInterface: DriveTaskInterface = object : DriveTaskInterface {
        override fun onFailure(t: Throwable?) {
            if (t != null && TextUtils.isEmpty(t.message)) {
                Log.e(TAG, "Load failure: ${t.message}")
                Toast.makeText(
                    this@FileListActivity,
                    "Load failure: ${t.message}",
                    Toast.LENGTH_SHORT
                ).show()
            }
        }

        override fun onResponse(files: List<File>?) {
            if (files.isNullOrEmpty()) {
                mBindingFileListActivity.noFileTextview.visibility = View.VISIBLE
            } else {
                mBindingFileListActivity.fileListView.adapter = FileListViewAdapter(files)
                mBindingFileListActivity.fileListView.layoutManager =
                    LinearLayoutManager(applicationContext)
            }
        }
    }

    companion object {
        private const val TAG: String = MainActivity.TAG
        const val INTENT_CREDENTIAL = "intent_credential"
    }
}